submissions/IhorOzerov/frogger/app.js

```diff
@@ -0,0 +1,113 @@
+const config = {
+    CANVAS_WIDTH : 505,
+    REFRESH_TIME : 500,
+    Y_START : 0,
+    X_START : 0,
+
+    ENEMY_CONST: {
+        refresh: -101,
+        locations: [62, 144, 228],
+        speed: (Math.random() * 200) + 200,
+        sprite: "images/enemy-bug.png"
+    },
+
+    PLAYER_CONST: {
+        x: 203,
+        y: 390,
+        stepX: 102,
+        stepY: 83,
+        sprite: "images/char-boy.png",
+        height: 60,
+        width: 70,
+        rightBorder: 400
+    }
+};
+
+const Enemy = function (x, y) {
+    this.x = x;
+    this.y = y;
+    this.speed = config.ENEMY_CONST.speed;
+    this.sprite = config.ENEMY_CONST.sprite;
+}; 
+
+Enemy.prototype.update = function (dt) {
+    this.x += config.ENEMY_CONST.speed * dt;
+    if (this.x > config.CANVAS_WIDTH) {
+        this.x = config.ENEMY_CONST.refresh;
+    };
+    this.checkCollision();
+};
+
+Enemy.prototype.checkCollision = function () {
+    
+    const enemyWidth = this.x + config.PLAYER_CONST.width;
+    const enemyHeight = this.y + config.PLAYER_CONST.height;
+    const playerWidth = player.x + config.PLAYER_CONST.width;
+    const playerHeight = player.y + config.PLAYER_CONST.height;
+    
+    if (player.x < enemyWidth &&
+        playerWidth > this.x &&
+        player.y < enemyHeight &&
+        playerHeight > this.y) {
+            player.x = config.PLAYER_CONST.x;
+            player.y = config.PLAYER_CONST.y;
+            }
+};
+
+Enemy.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const allEnemies = [];
+
+config.ENEMY_CONST.locations.forEach(
+    function (enemyPosition) {
+    const startPoint = Math.random()*config.CANVAS_WIDTH;
+    const enemy = new Enemy(startPoint, enemyPosition, config.ENEMY_CONST.speed);
+    allEnemies.push(enemy);
+});
+
+const Player = function() {
+    this.x = config.PLAYER_CONST.x;
+    this.y = config.PLAYER_CONST.y;
+    this.sprite = config.PLAYER_CONST.sprite;
+};
+const player = new Player;
+
+Player.prototype.update = function () {
+    if (this.y < config.Y_START) {
+        setTimeout(function () {
+            player.x = config.PLAYER_CONST.x;
+            player.y = config.PLAYER_CONST.y;
```

Comment:
Class definition should not refer a specific instance of the class.
There are means to refer to properties you actually use. Why this? (not a rhetoric question)

Reply:
I moved the character with coordinates and chose these two points for myself.

Reply:
i am renamed specific instance, please check.
i am not pretty sure am i understand

Reply:
am i need to make some constants like a math decision, but not like "i am decided that"?
or there is problem exactly in specific instance?

Reply:
Is "i decided that" equivalent to "i just guessed and it works"?
Calculations provide explanations. As a developer tasked to contribute to a project developed before by someone else (often previous contributors are unreachable as people tend to quit companies) you will want to understand what is going on in the code, why and what is behind numbers in particular.

Reply:
Second option will be closer to truth. 
After this fixes i am better see how prototype works and where it similar to class sintax. Thank you!